a3p-integration/proposals/z:acceptance/test-lib/sync-tools.test.js

```diff
@@ -0,0 +1,412 @@
+  async t => {
+    const { setValue, follow } = makeFakeFollow();
+    setValue({ status: {}, updated: 'balance' });
+
+    const waitP = waitUntilOfferResult(
+      'agoric12345',
+      'my-offer',
+      false,
+      { follow, setTimeout: globalThis.setTimeout },
+      {
+        maxRetries: 5,
+        retryIntervalMs: 1000,
+        log: t.log,
+        errorMessage: 'Wrong update type',
+      },
+    );
+
+    await t.throwsAsync(waitP);
+  },
+);
+
+test.serial('wait until offer result, wrong id - should throw', async t => {
+  const { setValue, follow } = makeFakeFollow();
+  setValue({ status: { id: 'your-offer' }, updated: 'offerStatus' });
+
+  const waitP = waitUntilOfferResult(
+    'agoric12345',
+    'my-offer',
+    false,
+    { follow, setTimeout: globalThis.setTimeout },
+    {
+      maxRetries: 5,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'Wrong offer id',
+    },
+  );
+
+  await t.throwsAsync(waitP);
+});
+
+test.serial('wait until offer result, no "status" - should throw', async t => {
+  const { setValue, follow } = makeFakeFollow();
+  setValue({ updated: 'offerStatus' });
+
+  const waitP = waitUntilOfferResult(
+    'agoric12345',
+    'my-offer',
+    false,
+    { follow, setTimeout: globalThis.setTimeout },
+    {
+      maxRetries: 5,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'No "status" object',
+    },
+  );
+
+  await t.throwsAsync(waitP);
+});
+
+test.serial(
+  'wait until offer result, numWantsSatisfied not equals to 1 - should throw',
+  async t => {
+    const { setValue, follow } = makeFakeFollow();
+    setValue({
+      status: { id: 'my-offer', numWantsSatisfied: 0 },
+      updated: 'offerStatus',
+    });
+
+    const waitP = waitUntilOfferResult(
+      'agoric12345',
+      'my-offer',
+      false,
+      { follow, setTimeout: globalThis.setTimeout },
+      {
+        maxRetries: 5,
+        retryIntervalMs: 1000,
+        log: t.log,
+        errorMessage: '"numWantsSatisfied" is not 1',
+      },
+    );
+
+    await t.throwsAsync(waitP);
+  },
+);
+
+test.serial('wait until offer result, do not wait for "payouts"', async t => {
+  const { setValue, follow } = makeFakeFollow();
+  setValue({ status: { id: 'my-offer' }, updated: 'offerStatus' });
+
+  const waitP = waitUntilOfferResult(
+    'agoric12345',
+    'my-offer',
+    false,
+    { follow, setTimeout: globalThis.setTimeout },
+    {
+      maxRetries: 7,
+      retryIntervalMs: 1000,
+      log: t.log,
+      errorMessage: 'offer not resulted on time',
+    },
+  );
+
+  setTimeout(
+    () =>
+      setValue({
+        status: { id: 'my-offer', numWantsSatisfied: 1 },
+        updated: 'offerStatus',
+      }),
+    1000,
+  ); // First, offer is seated
+  setTimeout(
+    () =>
+      setValue({
+        status: { id: 'my-offer', numWantsSatisfied: 1, result: 'thank you' },
+        updated: 'offerStatus',
+      }),
+    3000,
+  ); // First, offer is resulted
```

Comment:
This doesn't work grammatically. "First, offer is seated" works because "to be seated" is a valid verb, but there's no version of "result" that can act as a verb.
```suggestion
  ); // First offer got results
```
I think the comma is extraneous. This is referring to the "first offer", right?  With the comma, I can only read it as "the first thing that happens is that the offer is seated".

Reply:
resolved in https://github.com/Agoric/agoric-sdk/pull/10171/commits/d3f245300f2d42b83bc761c1e22950feeacd63db